### ISSUE TITLE
Sage test failures on Travis

### ISSUE THREAD

**Issue by asmeurer** [sympy/sympy#15802]
See the discussions at https://github.com/sympy/sympy/pull/15785. There are some sage failures intermittently on Travis like

```
SageMath version 8.3, Release Date: 2018-08-03
+sage -python bin/test sympy/external/tests/test_sage.py
Setting permissions of DOT_SAGE directory so only you can read and write it.
============================= test process starts ==============================
executable:         /home/travis/miniconda/envs/sage/bin/python3  (3.6.7-final-0) [CPython]
architecture:       64-bit
cache:              yes
ground types:       python 
numpy:              1.15.2
random seed:        33340000
hash randomization: on (PYTHONHASHSEED=1662812280)
------------------------------------------------------------------------
```

I think the build is timing out. 

I ran that command on a Linux docker container and it seems that no tests are actually run

```
(sage) root@812dbe06e1e0:~/sympy# sage -python bin/test sympy/external/tests/test_sage.py
Setting permissions of DOT_SAGE directory so only you can read and write it.
==================================================================== test process starts =====================================================================
executable:         /opt/conda/envs/sage/bin/python3  (3.6.7-final-0) [CPython]
architecture:       64-bit
cache:              yes
ground types:       python
numpy:              1.15.2
random seed:        45793267
hash randomization: on (PYTHONHASHSEED=256532873)

------------------------------------------------------------------------
(sage) root@812dbe06e1e0:~/sympy#
```

which is concerning. 

The next line fails like this

```
(sage) root@812dbe06e1e0:~/sympy#     PYTHONPATH=. sage -t sympy/external/tests/test_sage.py

Traceback (most recent call last):
  File "/opt/conda/envs/sage/bin/sage-runtests", line 125, in <module>
    from sage.doctest.control import DocTestController
  File "/opt/conda/envs/sage/lib/python3.6/site-packages/sage/doctest/control.py", line 33, in <module>
    from .sources import FileDocTestSource, DictAsObject
  File "/opt/conda/envs/sage/lib/python3.6/site-packages/sage/doctest/sources.py", line 33, in <module>
    from .parsing import SageDocTestParser
  File "/opt/conda/envs/sage/lib/python3.6/site-packages/sage/doctest/parsing.py", line 57, in <module>
    from sage.all import RealIntervalField
  File "/opt/conda/envs/sage/lib/python3.6/site-packages/sage/all.py", line 96, in <module>
    from sage.matrix.all     import *
  File "/opt/conda/envs/sage/lib/python3.6/site-packages/sage/matrix/__init__.py", line 2, in <module>
    import sage.matrix.args
  File "sage/matrix/args.pyx", line 25, in init sage.matrix.args
  File "/opt/conda/envs/sage/lib/python3.6/site-packages/sage/matrix/matrix_space.py", line 46, in <module>
    from . import matrix_modn_sparse
  File "sage/matrix/matrix_integer_dense.pxd", line 10, in init sage.matrix.matrix_modn_sparse
  File "sage/matrix/matrix_modn_dense_template_header.pxi", line 8, in init sage.matrix.matrix_integer_dense
ImportError: /opt/conda/envs/sage/lib/python3.6/site-packages/sage/matrix/../../../../liblinboxsage.so.0: undefined symbol: _ZNK6Givaro7IntegercvSsEv
```

If that's the same error that is happening on Travis, I don't know why it isn't showing up in the logs. It does look like an issue with the Sage package, though. 

CC @isuruf

**Comment by jksuom** [comment]
```
(sage) [jks@nocona sympy{master}]$ python
Python 3.6.7 | packaged by conda-forge | (default, Nov 21 2018, 03:09:43) 
[GCC 7.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from sympy.external.tests.test_sage import *
------------------------------------------------------------------------
Segmentation fault (core dumped)
```
With `python -v`, these are the last messages:
```
# extension module 'sage.matrix.matrix_mpolynomial_dense' loaded from '/home/jks/miniconda3/envs/sage/lib/python3.6/site-packages/sage/matrix/matrix_mpolynomial_dense.cpython-36m-x86_64-linux-gnu.so'
# extension module 'sage.matrix.matrix_mpolynomial_dense' executed from '/home/jks/miniconda3/envs/sage/lib/python3.6/site-packages/sage/matrix/matrix_mpolynomial_dense.cpython-36m-x86_64-linux-gnu.so'
import 'sage.matrix.matrix_mpolynomial_dense' # <_frozen_importlib_external.ExtensionFileLoader object at 0x7f4d482a29b0>
------------------------------------------------------------------------
Segmentation fault (core dumped)
```

**Comment by jksuom** [comment]
```
(sage) [jks@nocona sympy{master}]$ python
Python 3.6.7 | packaged by conda-forge | (default, Nov 21 2018, 03:09:43) 
[GCC 7.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from sympy.external import import_module
>>> sage = import_module('sage.all', __import__kwargs={'fromlist': ['all']})
------------------------------------------------------------------------
Segmentation fault (core dumped)
```

**Comment by jksuom** [comment]
This is a fragment of `python -v` log file (450 kB) at the point where importing of sage stops.
```
# extension module 'sage.matrix.matrix_rational_dense' loaded from '/home/jks/mi
niconda3/envs/sage/lib/python3.6/site-packages/sage/matrix/matrix_rational_dense
.cpython-36m-x86_64-linux-gnu.so'
# extension module 'sage.matrix.matrix_rational_dense' executed from '/home/jks/
miniconda3/envs/sage/lib/python3.6/site-packages/sage/matrix/matrix_rational_den
se.cpython-36m-x86_64-linux-gnu.so'
import 'sage.matrix.matrix_rational_dense' # <_frozen_importlib_external.Extensi
onFileLoader object at 0x7ff6af1f06d8>
# extension module 'sage.arith.multi_modular' loaded from '/home/jks/miniconda3/
envs/sage/lib/python3.6/site-packages/sage/arith/multi_modular.cpython-36m-x86_6
4-linux-gnu.so'
# extension module 'sage.arith.multi_modular' executed from '/home/jks/miniconda
3/envs/sage/lib/python3.6/site-packages/sage/arith/multi_modular.cpython-36m-x86
_64-linux-gnu.so'
import 'sage.arith.multi_modular' # <_frozen_importlib_external.ExtensionFileLoa
der object at 0x7ff6af1f0f98>
# destroy sage.matrix
# destroy sage.all
```

**Comment by asmeurer** [comment]
You can try python -X faulthandler to get a traceback.

**Comment by oscargus** [comment]
It seems to be the same thing happening at Travis. For unknown reasons, I got a traceback for my latest commit, see #15772 where I pasted it before seeing the more or less identical traceback here (also available from the test log file).

**Comment by asmeurer** [comment]
I opened an upstream issue on the feedstock https://github.com/conda-forge/sagelib-feedstock/issues/62. If we don't hear back on it soon we should consider temporarily moving the sage tests to a allowed failures build.

**Comment by jksuom** [comment]
I got nothing out of faulthandler, but this might give a hint:
```
(sage) [jks@nocona sympy{master}]$ python
Python 3.6.7 | packaged by conda-forge | (default, Nov 21 2018, 03:09:43) 
[GCC 7.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import sage.matrix.matrix_modn_dense_float
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/jks/miniconda3/envs/sage/lib/python3.6/site-packages/sage/matrix/__init__.py", line 2, in <module>
    import sage.matrix.args
  File "sage/matrix/args.pyx", line 25, in init sage.matrix.args
  File "/home/jks/miniconda3/envs/sage/lib/python3.6/site-packages/sage/matrix/matrix_space.py", line 43, in <module>
    from . import matrix_generic_dense
  File "sage/matrix/matrix_generic_dense.pyx", line 1, in init sage.matrix.matrix_generic_dense
  File "sage/matrix/matrix_dense.pyx", line 1, in init sage.matrix.matrix_dense
  File "sage/matrix/matrix2.pyx", line 1, in init sage.matrix.matrix2
  File "sage/matrix/matrix1.pyx", line 1, in init sage.matrix.matrix1
  File "sage/matrix/matrix0.pyx", line 30, in init sage.matrix.matrix0
  File "/home/jks/miniconda3/envs/sage/lib/python3.6/site-packages/sage/modules/free_module.py", line 167, in <module>
    from . import free_module_element
  File "sage/rings/integer.pxd", line 7, in init sage.modules.free_module_element
  File "sage/rings/rational.pxd", line 8, in init sage.rings.integer
  File "sage/rings/rational.pyx", line 95, in init sage.rings.rational
  File "sage/rings/real_mpfr.pyx", line 1, in init sage.rings.real_mpfr
  File "sage/rings/complex_number.pxd", line 6, in init sage.libs.mpmath.utils
  File "sage/rings/complex_double.pxd", line 10, in init sage.rings.complex_number
  File "sage/rings/complex_double.pyx", line 98, in init sage.rings.complex_double
ImportError: cannot import name complex_number
>>> 
```

**Comment by asmeurer** [comment]
I am trying @isuruf's workaround at https://github.com/sympy/sympy/pull/15821.

**Comment by asmeurer** [comment]
The workaround fixed the builds but let's leave this open so we can remember to remove it once conda forge is fixed.

**Comment by asmeurer** [comment]
So it looks like Travis won't pull in the latest master if you restart builds. I restarted https://travis-ci.org/sympy/sympy/builds/482008053 and the Sage job still fails. 

I guess everyone will need to push a new commit to each PR to get the builds working. If there are no changes that are required you can merge with master and push the merge commit. But please don't do this unless it is otherwise ready to merge, as we don't want to make the build queue too large.

**Comment by asmeurer** [comment]
Also presumably the default conda-forge sage will get fixed at some point, meaning those old jobs will pass again once they are restarted.

**Comment by mkoeppe** [comment]
This can be closed now that #21941 is merged. 
#21958 removes the old scripts for Sage integration testing.
